src/map/Map.js

```diff
@@ -520,43 +521,52 @@ L.Map = L.Evented.extend({
 
 	// private methods that modify map state
 
-	_resetView: function (center, zoom, preserveMapOffset, afterZoomAnim) {
+	_resetView: function (center, zoom) {
+		L.DomUtil.setPosition(this._mapPane, new L.Point(0, 0));
 
-		var zoomChanged = (this._zoom !== zoom);
+		var loading = !this._loaded;
+		this._loaded = true;
 
-		if (!afterZoomAnim) {
-			this.fire('movestart');
+		var zoomChanged = this._zoom !== zoom;
+		this
+			._moveStart(zoomChanged)
+			._move(center, zoom)
+			._moveEnd(zoomChanged);
 
-			if (zoomChanged) {
-				this.fire('zoomstart');
-			}
+		if (loading) {
+			this.fire('load');
 		}
+	},
 
-		this._zoom = zoom;
-		this._initialCenter = center;
-
-		if (!preserveMapOffset) {
-			L.DomUtil.setPosition(this._mapPane, new L.Point(0, 0));
+	_moveStart: function (zoomChanged) {
+		if (zoomChanged) {
+			this.fire('zoomstart');
 		}
+		return this.fire('movestart');
+	},
 
-		this._pixelOrigin = this._getNewPixelOrigin(center);
+	_move: function (center, zoom, data) {
+		if (zoom === undefined) {
+			zoom = this._zoom;
+		}
 
-		var loading = !this._loaded;
-		this._loaded = true;
+		var zoomChanged = this._zoom !== zoom;
 
-		this.fire('viewreset', {hard: !preserveMapOffset});

```

Comment:
Not firing `viewreset` means that the GridLayers' `_resetView()` method is not called, which means the grid origins are not updated, which means a grey map whenever a `setView()` call moves the map around without changing zoom level.

There must be a (clean) way to tell the grid layers to rebuild the level origin whenever the position of `L.Map._mapPane` is reset.

This is not exactly #3272, but the symptom is very, very similar.